core/bin/external_node/src/main.rs

```diff
@@ -507,11 +503,17 @@ async fn main() -> anyhow::Result<()> {
         .main_node_url()
         .expect("Main node URL is incorrect");
     tracing::info!("Main node URL is: {main_node_url}");
-    let main_node_client = <dyn MainNodeClient>::json_rpc(&main_node_url)
+    let main_node_client = L2Client::builder()
+        .with_rate_limit(
```

Comment:
with my suggested changes, we could just have with_allowed_requests_per_second here